src/alg_utils.jl

```diff
@@ -171,10 +171,22 @@ function DiffEqBase.prepare_alg(alg::Union{OrdinaryDiffEqAdaptiveImplicitAlgorit
     if typeof(alg) <: OrdinaryDiffEqImplicitExtrapolationAlgorithm
       return alg # remake fails, should get fixed
     else
-      remake(alg,chunk_size=Val{ForwardDiff.pickchunksize(x)}())
+      L = ArrayInterface.known_length(typeof(u0))
+      if L === nothing # dynamic sized
+        cs = ForwardDiff.pickchunksize(x)
+        remake(alg,chunk_size=cs)
+      else # statically sized
+        cs = pick_static_chunksize(Val{L}())
+        remake(alg,chunk_size=cs)
+      end
     end
 end
 
+@generated function pick_static_chunksize(::Val{chunksize}) where chunksize
+  x = ForwardDiff.pickchunksize(chunksize)
+  :(Val{$x}())
```

Comment:
FWIW, Pumas has been relying on const prop for this, and that seems to be fine so far.

Reply:
IIRC it caused a major performance change here 😅

Reply:
I should double check Pumas, then.